soda_machine.rb

```diff
@@ -7,12 +7,20 @@ def initialize(args = {})
   end
 
   def current_inventory_count
+    @sodas.size
   end
 
   def find_soda(soda_brand)
+    @sodas.find{|soda| soda.brand == soda_brand}
   end
 
   def sell(soda_brand)
+    this_soda = find_soda(soda_brand)
+    if this_soda != nil
+      @sodas.delete_at(@sodas.index(this_soda))
```

Comment:
well, that makes perfect sense.  good catch.  why not add a test for this and submit a pull request to the challenge writers (@b0uma, @mashalifshin, @salogel42)?

that way students who take this in the future will be forced to consider the difference between these methods